### PR TITLE
feat: respect ctrl+c & ctrl+a/ctrl+e for text nav in neo

### PR DESCRIPTION
**Pull request by dschaller**
## Summary

Improve `pulumi neo` line editing behavior so Ctrl+C clears an in-progress draft before arming the quit/cancel flow, while preserving Ctrl+A/Ctrl+E textarea navigation when editing text.

## Test plan

- [x] Added appropriate unit tests - For all changes
- [ ] Added a test in `pkg/engine/lifecycletest` - For all engine/protocol changes
- [ ] Added a conformance test in `pkg/testing/pulumi-test-language` - For language protocol changes
- [ ] Added a golden test in `pkg/backend/display` - For changes to the output renderers

## Validation

- [x] `make lint` — clean
- [ ] `make test_fast` — all pass
- [x] `make tidy_fix` — clean
- [x] `make format` — clean
- [ ] Relevant SDK tests pass (if SDK changes)
- [ ] `make check_proto` — clean (if proto changes)

## Changelog

- [x] Changelog entry added. If you do not believe this PR requires a changelog, ask a maintainer to apply
  the `impact/no-changelog-required` label.

## Risk

Low. This only affects `pulumi neo` TUI keyboard handling for draft input and mode-toggle shortcuts. The main risk is changing expected Ctrl+C/Ctrl+A behavior in edge cases, covered by focused unit tests. No public SDK API, engine lifecycle, protocol, or protobuf changes.

### CODE REVIEW

File: pkg/cmd/pulumi/neo/tui.go [hunk 1]
```diff
@@ -660,9 +695,25 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
...
+		// editing behavior users expect from agent CLIs. With an empty draft,
+		// it falls through to the quit/cancel gate below.
+		if keyStr == "ctrl+c" && m.textInput.Value() != "" {
+			m.textInput.Reset()
+			return m, nil
+		}
+
+		// When the user is editing a draft, let textarea keep ownership of its
+		// line-editing keymap before app-level shortcuts see the same key.
+		if m.draftEditingKey(msg) {
```
Comment (VenelinMartinov) [inline pkg/cmd/pulumi/neo/tui.go:708]: Does this mean that the CLI will consume this and it won't propagate to the terminal? I don't fully understand this bit - what behaviour does it change?

**Review by VenelinMartinov — APPROVED** [review]

File: pkg/cmd/pulumi/neo/tui.go (hunk 1 above)
Comment (dschaller) [reply pkg/cmd/pulumi/neo/tui.go:708]: Yeah - Bubble Tea has already received the key event from the terminal; this branch consumes it inside the TUI by handing it to the textarea and returning, so it won’t fall through to the app-level shortcuts below.

The user-visible change is for non-empty drafts: textarea editing bindings win over global shortcuts. In particular, Ctrl+A moves the cursor to the start of the input instead of cycling approval mode. With an empty draft, Ctrl+A still cycles approval mode as before. Same idea for other collisions, e.g. Ctrl+D deletes forward in a draft instead of arming quit, and Ctrl+C clears the draft before entering the quit/cancel path.
